qutebrowser/misc/sessions.py

```diff
@@ -239,6 +239,12 @@ def _save_tab(self, tab, active, with_history=True):
         for idx, item in enumerate(history):
             qtutils.ensure_valid(item)
             item_data = self._save_tab_item(tab, idx, item)
+
+            if not item.url().isValid():
```

Comment:
The conditional below this one does some juggling of the `active` flag, maybe have a think about whether that might be required here too.

Reply:
I was thinking more along the lines of maybe we should be doing the active flag juggling stuff in this case too? Or something like that? Like can we get a tab with an invalid current URL but a valid history?

Reply:
@toofar, what do you think?

Reply:
I haven't thought much about this issue, so I don't know if this can happen in other cases (eg an existing tab getting an invalid URL). I'm happy to go with the advice of the experts.